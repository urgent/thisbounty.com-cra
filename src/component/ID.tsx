import React, { ReactChildren } from 'react'
import styled from 'styled-components'

const Anchor = styled.a`
  font-size: 1.6rem;
  color: #fff;
  text-decoration: none;

  &:hover {
    text-decoration: underline;
  }
`

export function ID ({
  value,
  children
}: {
  value: number
  children?: ReactChildren
}) {
  return (
    <>
      <Anchor href={`#${value}`}>#{value}</Anchor>
      {children}
    </>
  )
}
