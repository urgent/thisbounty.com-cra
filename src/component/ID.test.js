import renderer from 'react-test-renderer';
import React from 'react'
import { ID } from './ID'

test('ID renders correctly', () => {
    const zero = renderer
        .create(<ID value={0} />)
        .toJSON();
    const id = renderer
        .create(<ID value={100} />)
        .toJSON();

    expect(zero).toMatchSnapshot();
    expect(id).toMatchSnapshot();
});