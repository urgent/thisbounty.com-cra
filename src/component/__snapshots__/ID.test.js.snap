// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ID renders correctly 1`] = `
<a
  className="sc-bdfBwQ cteGcY"
  href="#0"
>
  #
  0
</a>
`;

exports[`ID renders correctly 2`] = `
<a
  className="sc-bdfBwQ cteGcY"
  href="#100"
>
  #
  100
</a>
`;
